packages/babel-plugin-transform-es2015-modules-commonjs/src/index.js

```diff
@@ -339,69 +611,159 @@ export default function () {
+                if (!spec) {
+                  for (const [name, local] of imports[source].reexports.entries()) {
+                    if (name === "default") {
+                      topNodes.push(buildExportsFrom(t.stringLiteral(name), t.memberExpression(t.callExpression(this.addHelper("interopRequireDefault"), [uid]), local)));
+                    } else {
+                      topNodes.push(buildExportsFrom(t.stringLiteral(name), t.memberExpression(uid, local)));
+                    }
+                  }
+                } else {
+                  for (const [name, local] of imports[source].reexports.entries()) {
+                    hoistedExports.set(name, t.memberExpression(uid, local));
                   }
-                  wildcard = specifier.local;
-                } else if (t.isImportDefaultSpecifier(specifier)) {
-                  specifiers[i] = t.importSpecifier(specifier.local, t.identifier("default"));
                 }
               }
 
-              for (let specifier of specifiers) {
-                if (t.isImportSpecifier(specifier)) {
-                  let target = uid;
-                  if (specifier.imported.name === "default") {
-                    if (wildcard) {
-                      target = wildcard;
+              if (specImport) {
+                for (const specifier of specifiers) {
+                  if (t.isImportNamespaceSpecifier(specifier)) {
+                    remaps[specifier.local.name] = uid;
+                  } else if (t.isImportDefaultSpecifier(specifier)) {
+                    remaps[specifier.local.name] = t.memberExpression(uid, t.identifier("default"));
+                  } else if (specImport && t.isExportSpecifier(specifier)) {
+                    // do nothing
+                  } else {
+                    remaps[specifier.local.name] = t.memberExpression(uid, t.cloneWithoutLoc(specifier.imported));
+                  }
+                  if (specImport && !t.isExportSpecifier(specifier)) {
+                    remaps[specifier.local.name].nameSet = checkedImportMap.get(uid).nameSet;
+                  }
+                }
+              } else {
+                let wildcard;
+
+                for (let i = 0; i < specifiers.length; i++) {
+                  let specifier = specifiers[i];
+                  if (t.isImportNamespaceSpecifier(specifier)) {
+                    if (strict) {
+                      remaps[specifier.local.name] = uid;
                     } else {
-                      target = wildcard = path.scope.generateUidIdentifier(uid.name);
                       const varDecl = t.variableDeclaration("var", [
                         t.variableDeclarator(
-                          target,
+                          specifier.local,
                           t.callExpression(
-                            this.addHelper("interopRequireDefault"),
+                            this.addHelper("interopRequireWildcard"),
```

Comment:
git picked a very confusing diff for this block 😅 

This as not a string rename, the `interopRequireDefault` block was moved further down.